Lib/test/test_asyncio/test_queues.py

```diff
@@ -522,5 +522,221 @@ class PriorityQueueJoinTests(_QueueJoinTestMixin, unittest.IsolatedAsyncioTestCa
     q_class = asyncio.PriorityQueue
 
 
+class _QueueShutdownTestMixin:
+    q_class = None
+
+    @staticmethod
+    async def _ensure_started(task):
```

Comment:
> That this exists feels like a red flag

I am depending on a specific order of operations in the test (other arrangements  are covered by other tests or are trivial)

---

> Why is it a static method?

Minor style preference of mine, I'll remove it to make it consistent with the repo

---

> There's a 10 msec timeout, which is then ignored.

I didn't realise I could use a timeout of 0

> What if it times out?

I'm testing the status and result of the tasks later in the tests, so I'm ignoring it here

---

> why not just `await asyncio.sleep(0)`?

I didn't know `await asyncio.sleep(0)` would allow all other tasks to run, and instead had an explicit version. I'll do the replace you suggest and test it